### PR TITLE
Stack and its real life applications

### PR DESCRIPTION
**Pull request by dalalvarun**
## Stack and its real life applications

###Stack implementation using array
###Stack implementation using Linked List
###Multiple Stack
###Parenthesis Checker
###Infix to Postfix
###Infix to Prefix
###Prefix evaluation
###Postfix Evaluation

Write here what you have changed in codebase

### Self Check(Tick After Making pull Request)

- [ ] This issue was assigned to me if not it will marked as spam.
- [x] My file is in proper folder.
- [x] I am following clean code and Documentation.
- [x] I have added title and what program will do first in the file.

This wasn't assigned to me. But I have a PR lined up.

### CODE REVIEW

**Comment by div-bargali** [comment]
@dalalvarun Your implementation code is in the wrong directory.

File: C++/Data-Structures/Stacks/Evaluate Postfix Expression.cpp [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+ * Enter a postfix expression:2 3 1 * + 9 -
+ * 
+ * The evaluation is: -4
+ */
+
+int main()
+{
+    cout<<"Enter a postfix expression:";
+    char input[MAX];
+    gets(input);
```
Comment (hemanth-kotagiri) [inline C++/Data-Structures/Stacks/Evaluate Postfix Expression.cpp:30]: ``gets`` was not declared in this scope. Please fix this by including the required headers.

File: C/Data-Structures/Stacks/Evaluate Prefix Expression.cpp [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+                case '*':
+                    top = push(top,op1*op2);
+                    break;
+                case '/' :
+                    top = push(top,op1/op2);
+                    break;
+            }
+        }
+    }
+    cout<<"\nThe evaluation is: "<<peek(top);
```
Comment (hemanth-kotagiri) [inline C/Data-Structures/Stacks/Evaluate Prefix Expression.cpp:62]: Add an escape character ``\n`` or ``endl``.

File: C/Data-Structures/Stacks/Infix to Postfix Conversion.cpp [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+            while(peek(top)!='(')
+            {
+                char temp = peek(top);
+                cout<<temp<<" ";
+                top = pop(top);
+            }
+            top = pop(top);
+        }
+
+    }
```
Comment (hemanth-kotagiri) [inline C/Data-Structures/Stacks/Infix to Postfix Conversion.cpp:87]: Add a new line escape character or ``endl`` here as well.

File: C/Data-Structures/Stacks/Parenthesis Checker using Stacks.cpp [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+struct call_stack *top = nullptr;
+struct call_stack *push(char x);
+struct call_stack *pop();
+char peek();
+int main()
+{
+    cout<<"\nEnter an expression:";
+    char exp[MAX];
+    gets(exp);
```
Comment (hemanth-kotagiri) [inline C/Data-Structures/Stacks/Parenthesis Checker using Stacks.cpp:22]: ``gets`` is not declared in this scope.

**Review by hemanth-kotagiri — CHANGES_REQUESTED** [review]
Hello, @dalalvarun. Thank you for your PR.
Please consider fixing the issues that I addressed individually in the comments.
Format your code well in whichever file applicable.

Thanks!

**Pull request closed without merging**